### PR TITLE
brew: remove sudo call

### PR DESCRIPTION
**Pull request by iMichka**
Closes #5865

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

File: Library/Homebrew/brew.rb [hunk 1]
```diff
@@ -113,9 +113,7 @@ class MissingEnvironmentVariables < RuntimeError; end
 
     odie "Unknown command: #{cmd}" if !possible_tap || possible_tap.installed?
 
-    brew_uid = HOMEBREW_BREW_FILE.stat.uid
     tap_commands = []
-    tap_commands += %W[/usr/bin/sudo -u ##{brew_uid}] if Process.uid.zero? && !brew_uid.zero?
     # Unset HOMEBREW_HELP to avoid confusing the tap
     ENV.delete("HOMEBREW_HELP") if help_flag
     tap_commands += %W[#{HOMEBREW_BREW_FILE} tap #{possible_tap}]
```
Comment (sjackman) [inline Library/Homebrew/brew.rb]:
```suggestion
    safe_system(HOMEBREW_BREW_FILE, "tap", possible_tap)
```

File: Library/Homebrew/brew.rb [hunk 2]
```diff
@@ -113,9 +113,7 @@ class MissingEnvironmentVariables < RuntimeError; end
 
     odie "Unknown command: #{cmd}" if !possible_tap || possible_tap.installed?
 
-    brew_uid = HOMEBREW_BREW_FILE.stat.uid
     tap_commands = []
```
Comment (sjackman) [inline Library/Homebrew/brew.rb]:
```suggestion
```

**Comment by sjackman** [comment]
```
+ sudo -E -u linuxbrew /home/linuxbrew/.linuxbrew/bin/brew test-bot
Error: Invalid tap name '{"name"=>"homebrew/test-bot", "user"=>"Homebrew", "repo"=>"test-bot", "path"=>"/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-test-bot", "installed"=>false, "official"=>true, "formula_names"=>[], "formula_files"=>[], "command_files"=>[], "pinned"=>false}'
Error: Failure while executing; `/home/linuxbrew/.linuxbrew/bin/brew tap \{\"name\"=\>\"homebrew/test-bot\",\ \"user\"=\>\"Homebrew\",\ \"repo\"=\>\"test-bot\",\ \"path\"=\>\"/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-test-bot\",\ \"installed\"=\>false,\ \"official\"=\>true,\ \"formula_names\"=\>\[\],\ \"formula_files\"=\>\[\],\ \"command_files\"=\>\[\],\ \"pinned\"=\>false\}` exited with 1.
```
Odd. I guess it needs `possible_tap.name`. Odd that it worked before. `#{possible_tap}` may be equivalent to `possible_tap.to_s`. It looks like we're getting `possible_tap.inspect` for some reason now.

File: Library/Homebrew/brew.rb [hunk 3]
```diff
@@ -113,13 +113,9 @@ class MissingEnvironmentVariables < RuntimeError; end
...
     odie "Unknown command: #{cmd}" if !possible_tap || possible_tap.installed?
 
-    brew_uid = HOMEBREW_BREW_FILE.stat.uid
-    tap_commands = []
-    tap_commands += %W[/usr/bin/sudo -u ##{brew_uid}] if Process.uid.zero? && !brew_uid.zero?
     # Unset HOMEBREW_HELP to avoid confusing the tap
     ENV.delete("HOMEBREW_HELP") if help_flag
-    tap_commands += %W[#{HOMEBREW_BREW_FILE} tap #{possible_tap}]
-    safe_system(*tap_commands)
+    safe_system HOMEBREW_BREW_FILE, "tap", possible_tap
```
Comment (sjackman) [inline Library/Homebrew/brew.rb]:
```suggestion
    safe_system HOMEBREW_BREW_FILE, "tap", possible_tap.name
```

**Comment by iMichka** [comment]
> Odd. I guess it needs possible_tap.name. Odd that it worked before. #{possible_tap} may be equivalent to possible_tap.to_s. It looks like we're getting possible_tap.inspect for some reason now.

That's because each time I try to change something in brew, I stumble over weird things :)

Anyway, it is better to call `.name`, instead of relying on the `.to_s` implementation which may change.

**Comment by MikeMcQuaid** [comment]
Good call, thanks @iMichka!

**Comment by sjackman** [comment]
This PR was reverted in PR #6010.

**Comment by iMichka** [comment]
cc @jonchang

**Pull request merged**